scio-core/src/main/scala/com/spotify/scio/nio/TextIO.scala

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2016 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio.nio
+
+import com.spotify.scio.ScioContext
+import com.spotify.scio.io.{FileStorage, Tap}
+import com.spotify.scio.util.ScioUtil
+import com.spotify.scio.values.SCollection
+import org.apache.beam.sdk.io.{Compression, FileBasedSink, TextIO => BTextIO}
+
+import scala.concurrent.Future
+
+case class TextIO(name: String, path: String) extends ScioIO[String] with Tap[String] {
```

Comment:
added name as a parameter, not sure this is a good idea in general, but in streaming mode, the name is super useful when doing an update later, where you have to map transformed operations to make it works.